CodeLuau/Speaker.cs

```
@@ -4,162 +4,123 @@
+            var domains = new List<string> {"aol.com", "prodigy.com", "compuserve.com"};
+
+            if (string.IsNullOrWhiteSpace(FirstName))
+            {
+                return new RegisterResponse(RegisterError.FirstNameRequired);
+            }
+
+            if (string.IsNullOrWhiteSpace(LastName))
+            {
+                return new RegisterResponse(RegisterError.LastNameRequired);
+            }
+
+            if (string.IsNullOrWhiteSpace(Email))
+            {
+                return new RegisterResponse(RegisterError.EmailRequired);
+            }
+
+            var employers = new List<string>() {"Pluralsight", "Microsoft", "Google"};
+            var isValid = Experience > 10 || HasBlog || Certifications.Count() > 3 ||
+                                  employers.Contains(Employer);
+
+            if (!isValid)
+            {
+                isValid = Email_IsValid(domains, isValid);
+            }
+
+            if (!isValid)
+            {
+                return new RegisterResponse(RegisterError.SpeakerDoesNotMeetStandards);
+            }
+
+            if (Sessions.Count == 0)
+            {
+                return new RegisterResponse(RegisterError.NoSessionsProvided);
+            }
+
+            isApproved = Session_IsApproved(technologies, isApproved);
+
+            if (isApproved == false)
+            {
+                return new RegisterResponse(RegisterError.NoSessionsApproved);
+            }
+
+            RegistrationFee = Experience <= 1 ? 500
+                : Experience >= 2 && Experience <= 3 ? 250
+                : Experience >= 4 && Experience <= 5 ? 100
+                : Experience >= 6 && Experience <= 9 ? 50
+                : 0;
+            try
+            {
+                speakerId = repository.SaveSpeaker(this);
+            }
+            catch (Exception e)
+            {
+                //in case the db call fails 
+            }
```

This method should not hide the DB error. Just let the error bubble up to upper levels.